### PR TITLE
Deprecate request handlers

### PR DESCRIPTION
**Pull request by kian-thompson**
### requestHandlers are deprecated

The concept of `requestHandlers` has been deprecated. Please migrate all usage of the following APIs to the new `entryPoint` pattern:

-   `requestHandler` property in `ContainerRuntime.loadRuntime(...)`
-   `RuntimeRequestHandler`
-   `RuntimeRequestHandlerBuilder`
-   `defaultFluidObjectRequestHandler(...)`
-   `defaultRouteRequestHandler(...)`
-   `mountableViewRequestHandler(...)`
-   `buildRuntimeRequestHandler(...)`
-   `createFluidObjectResponse(...)`
-   `handleFromLegacyUri(...)`
-   `rootDataStoreRequestHandler(...)`

See [Removing-IFluidRouter.md](https://github.com/microsoft/FluidFramework/blob/main/packages/common/core-interfaces/Removing-IFluidRouter.md) for more details.

[AB#5375](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/5375)

### CODE REVIEW

File: examples/apps/attributable-map/src/modelContainerRuntimeFactoryWithAttribution.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ import {
 	IProvideRuntimeAttributor,
 } from "@fluid-experimental/attributor";
 import { FluidObject } from "@fluidframework/core-interfaces";
+// eslint-disable-next-line import/no-deprecated
```
Comment (markfields) [inline examples/apps/attributable-map/src/modelContainerRuntimeFactoryWithAttribution.ts:21]: Do we have a task open to clean all these examples up?

File: packages/framework/aqueduct/src/data-objects/pureDataObject.ts [hunk 2]
```diff
@@ -108,7 +108,9 @@ export abstract class PureDataObject<I extends DataObjectTypes = DataObjectTypes
 	 * 2. the request url is empty
 	 */
 	public async request(req: IRequest): Promise<IResponse> {
-		return defaultFluidObjectRequestHandler(this, req);
+		return req.url === "" || req.url === "/" || req.url.startsWith("/?")
+			? { mimeType: "fluid/object", status: 200, value: this }
+			: create404Response(req);
 	}
 
 	// #endregion IFluidRouter
```
Comment (markfields) [inline packages/framework/aqueduct/src/data-objects/pureDataObject.ts:116]: Clean-up later: Remove these `#region` comments

File: packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts [hunk 3]
```diff
@@ -59,12 +59,12 @@ async function createDataObject<
 	// request mixin in
 	runtimeClass = mixinRequestHandler(
 		async (request: IRequest, runtimeArg: FluidDataStoreRuntime) => {
-			const router: FluidObject<IFluidRouter> = await runtimeArg.entryPoint.get();
+			const router = (await runtimeArg.entryPoint.get()) as TObj;
```
Comment (markfields) [inline packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts:62]: nit: rename variable to dataObject

File: packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts [hunk 4]
```diff
@@ -59,12 +59,12 @@ async function createDataObject<
...
 			assert(
-				router.IFluidRouter !== undefined,
-				0x469 /* Data store runtime entryPoint is not an IFluidRouter */,
+				router.request !== undefined,
```
Comment (markfields) [inline packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts:64]: Don't do it but it's tempting to update this to be `router instanceof(PureDataObject)`

File: packages/framework/request-handler/src/requestHandlers.ts [hunk 5]
```diff
@@ -92,6 +95,7 @@ class LegacyUriHandle<T = FluidObject & IFluidLoadable> implements IFluidHandle<
 	}
 }
 
+/** @deprecated - Will be removed in future major release. Migrate all usage of IFluidRouter to the "entryPoint" pattern. Refer to Removing-IFluidRouter.md */
 export function handleFromLegacyUri<T = FluidObject & IFluidLoadable>(
```
Comment (markfields) [inline packages/framework/request-handler/src/requestHandlers.ts:99]: Wow never heard of this.

**Review by markfields — APPROVED** [review]
LGTM! Exciting!

File: examples/apps/attributable-map/src/modelContainerRuntimeFactoryWithAttribution.ts (hunk 1 above)
Comment (kian-thompson) [reply examples/apps/attributable-map/src/modelContainerRuntimeFactoryWithAttribution.ts:21]: [AB#4990](https://dev.azure.com/fluidframework/internal/_workitems/edit/4990)

File: packages/framework/aqueduct/src/data-objects/pureDataObject.ts (hunk 2 above)
Comment (kian-thompson) [reply packages/framework/aqueduct/src/data-objects/pureDataObject.ts:116]: Agreed. Pretty sure these are for collapsing areas of code in Visual Studio.

**Pull request merged**